### PR TITLE
feat(api): support adamic-adar & resource-allocation algorithms

### PR DESCRIPTION
**Pull request by imbajin**
TODO:
- [ ]  add test in client side

### CODE REVIEW

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/AdamicAdarAPI.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import com.baidu.hugegraph.type.define.Directions;
+import com.baidu.hugegraph.util.E;
+import com.baidu.hugegraph.util.JsonUtil;
+import com.codahale.metrics.annotation.Timed;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * This API include similar prediction algorithms, now include:
+ *  - Adamic Adar
+ *  - Resource Allocation
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/AdamicAdarAPI.java:50]: also update comment

File: hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/ResourceAllocationAPI.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import com.baidu.hugegraph.traversal.algorithm.PredictionTraverser;
+import com.baidu.hugegraph.type.define.Directions;
+import com.baidu.hugegraph.util.E;
+import com.baidu.hugegraph.util.JsonUtil;
+import com.codahale.metrics.annotation.Timed;
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * This API include similar prediction algorithms, now include:
+ *  - Adamic Adar
```
Comment (javeme) [inline hugegraph-api/src/main/java/com/baidu/hugegraph/api/traversers/ResourceAllocationAPI.java:49]: ditto

File: hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/PredictionTraverser.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    public double resourceAllocation(Id source, Id target, Directions dir,
+                                     String label, long degree, long limit) {
+        Set<Id> neighbors = checkAndGetCommonNeighbors(source, target, dir,
+                                                       label, degree, limit);
+        EdgeStep step = label == null ? new EdgeStep(graph(), dir) :
+                        new EdgeStep(graph(), dir, ImmutableList.of(label));
+
+        double sum = 0.0;
+        for (Id vid : neighbors) {
+            sum += 1.0 / this.edgesCount(vid, step);
```
Comment (javeme) [inline hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/PredictionTraverser.java:60]: may this.edgesCount() return 0

Comment (imbajin) [reply hugegraph-core/src/main/java/com/baidu/hugegraph/traversal/algorithm/PredictionTraverser.java:60]: why common neighbor vertex's degree could be zero? seems at least is 1 or 2?

**Review by javeme — APPROVED** [review]

**Review by coderzc — APPROVED** [review]

**Pull request merged**